pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -1066,43 +1058,146 @@ fn choose_exec_method(privdata: &PrivateData) -> ExecMethodType {
     }
 }
 
-fn assign_exec_method(custom_state: &mut PdbScanState) {
-    match &custom_state.exec_method_type {
-        ExecMethodType::Normal => custom_state.assign_exec_method(NormalScanExecState::default()),
+///
+/// Creates and assigns the execution method which was chosen at planning time.
+///
+/// TODO: This method currently has a fallback to the `Normal` execution mode for rare cases where
+/// the execution time target list actually contains _fewer_ fast fields than we expected at
+/// execution time. See #TODO-ISSUE-NUMBER
```

Comment:
Calling this out specifically. It's fairly rare, because all of the unit tests (except one: see my other comment) and proptests pass without the fallback. But we might want to expand the fallback to be defensive against assertion failures due to the change in behavior.

Reply:
This comment still needs some revision: sorry, it was late.

The case that is rare is that we fail to extract a set of fast fields from the execution time target list which contains _enough_ fields to populate the full target list. i.e. that the execution time tuple/target-list is wider than the execution-time fast fields.